File: commands/Fun/reverse.js
const { Command } = require("klasa");
const { clean } = require("../../utils/utils.js");

class Reverse extends Command {
  constructor(...args) {
    super(...args, {
      description: "Reverses your text.",
      usage: "<text:string>",
      aliases: ["rev"]
    });
  }
  
  async run(msg, [text]) {
    return msg.send(clean(msg, text.split("").reverse().join("")));
  }
}

module.exports = Reverse;

File: commands/fun/reverse.go
package fun

import (
  "strings"
  "github.com/sapphire-cord/sapphire"
)

// Reverses a given input
// Usage: <text...>
// Aliases: rev
func Reverse(ctx *sapphire.CommandContext) {

  var chars []string = strings.Split(ctx.JoinedArgs(), "");

  for i, j := 0, len(chars)-1; i < j; i, j = i+1, j-1 {
    chars[i], chars[j] = chars[j], chars[i]
  }

  ctx.Reply(sapphire.Escape(strings.Join(chars, "")))
}
